src/components/position_details_modal/position_details_modal.tsx

```diff
@@ -64,21 +71,111 @@ const PositionDetailsModal = ({
     setStopLossToggle(bool);
   };
 
-  const getSlToggleFunction = (slToggleFunction: () => void) => {
-    slToggleFunction();
+  const getTpValue = (value: number) => {
+    setTakeProfitValue(value);
+    // console.log('tp value from Trading Input:', value);
   };
 
+  const getSlValue = (value: number) => {
+    setStopLossValue(value);
+    // console.log('sl value from Trading Input:', value);
+  };
+
+  // const getSlToggleFunction = (slToggleFunction: () => void) => {
+  //   slToggleFunction();
+  // };
+
+  // TODO: i18n
+  const displayedTypeOfPosition =
+    openCfdDetails?.typeOfPosition === 'BUY' ? 'Up (Buy)' : 'Down (Sell)';
+
+  const displayedPnLColor =
+    openCfdDetails?.pnl.type === 'UP'
+      ? PNL_COLOR_TYPE.profit
+      : openCfdDetails?.pnl.type === 'DOWN'
+      ? PNL_COLOR_TYPE.loss
+      : PNL_COLOR_TYPE.equal;
+
+  const displayedBorderColor =
+    openCfdDetails?.typeOfPosition === 'BUY' ? BORDER_COLOR_TYPE.long : BORDER_COLOR_TYPE.short;
+
   const isDisplayedTakeProfitSetting = takeProfitToggle ? 'flex' : 'invisible';
-  const isDisplayedStopLossSetting = stopLossToggle || disabledSlToggle ? 'flex' : 'invisible';
+  const isDisplayedStopLossSetting = stopLossToggle ? 'flex' : 'invisible';
+
+  const displayedSlLowerLimit = openCfdDetails?.guranteedStop
+    ? openCfdDetails?.stopLoss ?? openCfdDetails.recommendedSl
+    : slLowerLimit;
+  const displayedSlUpperLimit = openCfdDetails?.guranteedStop
+    ? openCfdDetails?.stopLoss ?? openCfdDetails.recommendedSl
+    : slUpperLimit;
+
+  const buttonClickHandler = () => {
+    // console.log('btn clicked');
+
+    let changedProperties = {};
+    let takeProfit = {};
+    let stopLoss = {};
+
+    // Detect if tpValue has changed
+    if (takeProfitToggle && takeProfitValue !== openCfdDetails.takeProfit) {
```

Comment:
- Remove `toggle` properties, unnecessary to pass `toggle` to Metamask
- Turn off `takeProfitToggle`, pass `takeProfitAmount=0` to cancel the take-profit